src/core/ddsc/src/dds_topic.c

```diff
@@ -871,32 +871,47 @@ static dds_entity_t find_local_topic_impl (dds_find_scope_t scope, dds_participa
 
 #ifdef DDS_HAS_TOPIC_DISCOVERY
 
-static dds_entity_t find_remote_topic_impl (dds_participant *pp_topic, const char *name, dds_duration_t timeout)
+static dds_entity_t find_remote_topic_impl (dds_participant *pp_topic, const char *name, const dds_typeinfo_t *type_info, dds_duration_t timeout)
 {
   // On entry: pp_topic is pinned, no locks held
 
   dds_entity_t ret;
   struct ddsi_topic_definition *tpd;
-  struct ddsi_sertype *sertype;
+  struct ddsi_domaingv * gv = &pp_topic->m_entity.m_domain->gv;
+  const struct ddsi_typeid *type_id = ddsi_typeinfo_complete_typeid (type_info);
+  struct ddsi_type *resolved_type = NULL;
 
-  if ((ret = lookup_topic_definition_by_name (&pp_topic->m_entity.m_domain->gv, name, &tpd)) != DDS_RETCODE_OK)
+  if ((ret = lookup_topic_definition (gv, name, type_id, &tpd)) != DDS_RETCODE_OK)
     return ret;
   if (tpd == NULL)
     return DDS_RETCODE_OK;
-  if ((ret = dds_resolve_type (pp_topic->m_entity.m_hdllink.hdl, ddsi_type_pair_complete_id (tpd->type_pair), timeout, &sertype)) != DDS_RETCODE_OK)
+
+  const struct ddsi_typeid *tpd_type_id = ddsi_type_pair_complete_id (tpd->type_pair);
+  assert (!type_info || !ddsi_typeid_compare (tpd_type_id, type_id));
+  if (!ddsi_type_resolved (gv, tpd->type_pair->complete, DDSI_TYPE_INCLUDE_DEPS))
```

Comment:
This condition seems to add little of value, but perhaps I am overlooking something?

Reply:
You're right, fixed in [9eb81bc](https://github.com/eclipse-cyclonedds/cyclonedds/pull/1308/commits/9eb81bc832506d2dac3297ddb603fffe440d7879)